### DISCUSSION TITLE
CAN and LIN logging to the same MF4 file

### DISCUSSION THREAD

**Discussion by MatRosiek** [danielhrisca/asammdf#958] [Q&A]
Hello,
Im struggling to write some code in Python which will allow to save `CAN` messages and `LIN` messages to the same MF4 file.
Im a little bit lost of how I should do it via this library. 

1. How to actually log CAN frame for example to MF4 file? I stumbled upon this example here: [python-can](https://python-can.readthedocs.io/en/stable/_modules/can/io/mf4.html#MF4Writer). 
2. Why there is actually `'CAN_DataFrame'` fixed string needed? 
3. What if i want to group data in such file like for example having two groups called `LIN` and second one `CAN` how to do it? 
4. Suppose if I have 2 `CAN` interfaces and I need to log messages to this one file - is it possible? How should i group data from `CAN0` and e.g. `CAN1` in this file?

Im a little bit confused with usage of `Signal` class. As it looks like it can handle grouping too?

Thanks in advance for any help.

**Comment by danielhrisca** [comment]
> 2\. Why there is actually `'CAN_DataFrame'` fixed string needed?

This is the MDF spec for bus logging. You need to respect the field names and also to set the proper metadata in the file. Use the python-can code as a guide. You can have multiple groups with the same channel names; for example you can group the logging using the bus channel number (0, 1, ...) by setting the `BusChannel` value. Have a look here https://github.com/danielhrisca/asammdf/files/4076869/J1939-DBC-MDF4.zip

**Reply by dsyue** [reply]
thank you for your answer, i am searching for it!
